src/libsyntax/parse/token.rs

```diff
@@ -323,6 +323,22 @@ impl Token {
         }
     }
 
+    /// Returns `true` if token is special identifier in module name
+    #[allow(non_upper_case_globals)]
+    pub fn is_special_ident_in_modname(&self) -> bool {
+        match *self {
+            Ident(sid, ModName) => {
+                let n = sid.name;
+
+                   n == SELF_KEYWORD_NAME
+                || n == STATIC_KEYWORD_NAME
+                || n == SUPER_KEYWORD_NAME
+                || n == SELF_TYPE_KEYWORD_NAME
```

Comment:
I would prefer we just check `SUPER_KEYWORD_NAME` and `SELF_KEYWORD_NAME` here so the errors are as consistent as possible with other paths.